src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3244,61 +3272,74 @@ public static void createA1A6ToStaticNets(Design design) {
-
                 BEL lut6Bel = (fiveOrSix == '5') ? si.getBEL(belName.charAt(0) + "6LUT") : bel;
                 Net a6Net = si.getNetFromSiteWire(lut6Bel.getPin("A6").getSiteWireName());
 
-                // SRL16Es that have been transformed from SRLC32E require GND on their A6 pin
-                if (cell.getType().equals("SRL16E") && "SRLC32E".equals(cell.getPropertyValueString("XILINX_LEGACY_PRIM"))) {
-                    staticNet = gndNet;
-                    // Expect sitewire to be VCC and GND
-                    if (!a6Net.isStaticNet()) {
-                        throw new RuntimeException("ERROR: Site pin " + si.getSiteName() + "/" + belName.charAt(0) + "6 is not a static net");
+                boolean expectGndNet = false;
+                if ("SRL16E".equals(cell.getType())) {
+                    String pinName = belName.charAt(0) + "1";
+                    SitePinInst spi = si.getSitePinInst(pinName);
+                    if (spi == null) {
+                        vccNet.createPin(pinName, si);
                     }
-                } else {
-                    // Tie A6 to staticNet only if sitewire says so
-                    if (a6Net != staticNet) {
-                        continue;
+
+                    // SRL16Es that have been transformed from SRLC32E require GND on their A6 pin
+                    if ("SRLC32E".equals(cell.getPropertyValueString("XILINX_LEGACY_PRIM"))) {
+                        expectGndNet = true;
+                        staticNet = gndNet;
+                        // Expect sitewire to be VCC or GND
+                        if (!a6Net.isStaticNet()) {
+                            throw new RuntimeException("ERROR: Site pin " + si.getSiteName() + "/" + belName.charAt(0) + "6 is not a static net");
+                        }
                     }
                 }
 
+                // Tie A6 to staticNet only if sitewire says so
+                if (a6Net != staticNet && !expectGndNet) {
+                    continue;
+                }
+
                 if (cell.getLogicalPinMapping("O5") != null) {
                     // LUT output comes out on O5
-                    if (fiveOrSix == '5') {
-                        // It's a 5LUT
-                        if (si.getCell(belName.charAt(0) + "6LUT") != null) {
-                            // But 6LUT exists; let the 6LUT deal with it
-                            continue;
-                        }
-                    } else {
-                        throw new RuntimeException("Assumption that only 5LUTs can use O5 failed here.");
-                    }
+                    assert(fiveOrSix == '5');
                 } else {
                     if (fiveOrSix != '6') {
                         // Assume that O6 is only driven by 6LUT, even though possible for 5LUT, unless
                         // it's a routethru
-                        assert (cell.isRoutethru());
+                        assert(cell.isRoutethru());
```

Comment:
Our coding guidelines does not stipulate a space or absence thereof.  Since `assert` is a keyword rather than a method, it seems like my IDE follows the same as `while` and `for`.  Probably not worth legislating, but I think its one of those inconsistencies that has leaked through.

Reply:
Mine likes to add a space too, but I personally prefer not one which is the `C` way. Let's restore it to prevent a diff.